### ISSUE TITLE
Wrong calculation of validation loss

### ISSUE THREAD

**Issue by ATrefzer** [udacity/deep-learning-v2-pytorch#129]
I think there is a mistake in the validation code in the notebook:
mnist-mlp/mnist_mlp_solution_with_validation.ipynb

Your code:

\# print training/validation statistics
\# calculate average loss over an epoch
train_loss = train_loss/len(train_loader.dataset)
valid_loss = valid_loss/len(valid_loader.dataset)

Note that you divide both train_loss and valid_loss by 60000. Both loaders uses the train data set.
You should probably use:

train_loss = train_loss / len(train_sampler) \# 48000
valid_loss = valid_loss /len(valid_sampler) \#12000

**Comment by kuzand** [comment]
Yeah, the same mistake is in cifar-cnn notebook. The wrong way:
```
# calculate average losses
train_loss = train_loss/len(train_loader.dataset)
valid_loss = valid_loss/len(valid_loader.dataset)
```
where len(train_loader.dataset) = len(valid_loader.dataset) = 50000.

A correct one is:
```
train_loss = train_loss/(len(train_loader) * batch_size)
valid_loss = valid_loss/(len(valid_loader) * batch_size)
```

**Comment by cezannec** [comment]
Thank you for noticing, pushing fixes, now!

**Comment by marcindulak** [comment]
There still an instance of improper averaging https://github.com/udacity/deep-learning-v2-pytorch/blob/6ba1dff44a941bd46afa8a4c2246ef090981471d/convolutional-neural-networks/cifar-cnn/cifar10_cnn_exercise.ipynb?short_path=ff52321#L350-L352
